web/client/components/buttons/ZoomToMaxExtentButton.jsx

```diff
@@ -0,0 +1,100 @@
+ *  - style: {object}         a css-like hash to define the style on the component
+ *  - glyphicon: {string}     bootstrap glyphicon name
+ *  - text: {string|element}  text content for the button
+ *  - btnSize: {string}       bootstrap button size ('large', 'small', 'xsmall')
+ */
+var ZoomToMaxExtentButton = React.createClass({
+    propTypes: {
+        id: React.PropTypes.string,
+        style: React.PropTypes.object,
+        glyphicon: React.PropTypes.string,
+        text: React.PropTypes.string,
+        btnSize: React.PropTypes.oneOf(['large', 'medium', 'small', 'xsmall']),
+        // redux store slice with map configuration (bound through connect to store at the end of the file)
+        mapConfig: React.PropTypes.object,
+        // redux store dispatch func
+        dispatch: React.PropTypes.func
+    },
+    getDefaultProps() {
+        return {
+            id: "mapstore-zoomtomaxextent",
+            style: undefined,
+            glyphicon: "resize-full",
+            text: undefined,
+            hiddenText: true,
+            btnSize: 'xsmall'
+        };
+    },
+    render() {
+        return (
+            <Button
+                id={this.props.id}
+                bsStyle="default"
+                bsSize={this.props.btnSize}
+                onClick={() => this.zoomToMaxExtent()}>
+                {this.props.glyphicon ? <Glyphicon glyph={this.props.glyphicon}/> : null}
+                {this.props.glyphicon && this.props.text ? "\u00A0" : null}
+                {this.props.text}
+            </Button>
+        );
+    },
+    zoomToMaxExtent() {
+        var mapConfig = this.props.mapConfig;
+        var maxExtent = mapConfig.maxExtent;
+        var bbox;
+
+        if (maxExtent &&
+            Object.prototype.toString.call(maxExtent) === '[object Array]') {
+            // zoom map to the max. extent defined in the map's config
+            bbox = {
+                bounds: {
+                    minx: maxExtent[0],
+                    miny: maxExtent[1],
+                    maxx: maxExtent[2],
+                    maxy: maxExtent[3]
+                },
+                crs: mapConfig.projection,
+                rotation: 0
+            };
+            // adapt the map view by dispatching to the corresponding action
+            this.props.dispatch(changeMapView(this.props.mapConfig.center, -1,
+                bbox, this.props.mapConfig.size));
+        } else {
+            // zoom to zoom level 1 as fallback if no max extent is defined
+
+            // adapt the map view by dispatching to the corresponding action
+            this.props.dispatch(changeMapView(this.props.mapConfig.center, 1,
+                this.props.mapConfig.maxExtent, this.props.mapConfig.size));
+        }
+
+    }
+});
+
+// In addition to the state, 'connect' puts 'dispatch' in our props.
+// connect Redux store slice with map configuration
+module.exports = connect((state) => {
```

Comment:
It would be preferrable to bind properties in the plugins index.js as we do for other plugins, instead of connecting directly to the store